### ISSUE TITLE
(PDOM) - Consider benefits of new list design pattern that doesn't read bullets

### ISSUE THREAD

**Issue by terracoda** [phetsims/gravity-force-lab-basics#289]
In https://github.com/phetsims/a11y-research/issues/158, we identified a potentially nice way to code lists.

This sim has a lot of lists.

For the upcoming re-release of this sim with Voicing, would it be too much trouble to try this list style for this sim's PDOM?

**Comment by jessegreenberg** [comment]
Discussed 7/20/21:

The fix proposed in https://github.com/phetsims/a11y-research/issues/158 should be quick to apply, I will go ahead with it. The hope is that it will sound at least as good as what we have now.

**Comment by jessegreenberg** [comment]
Work was done in scenery to support this, we are ready to try it out in the sim.

**Comment by jessegreenberg** [comment]
Done in the above commit, I confirmed that `list-style` was set to `none` on the list items. I also confirmed that it removed the semantic information from the items when using NVDA, NVDA no longer says "bullet" when going through the list items.

@terracoda would you like to review this in VoiceOver?

**Comment by terracoda** [comment]
Yes, reviewing :-)

**Comment by terracoda** [comment]
@jessegreenberg, the lists under the mass sphere headings sound different than the list in the scene summary.

In both cases I no longer hear "bullet, bullet" which is awesome.
I hear the number of list items in the mass description lists, but not in the scene summary.

Is something different in the code?

**Comment by jessegreenberg** [comment]
Ah yes, only the list in the scene summary is using this new pattern. Will apply to the new lists too.

**Comment by jessegreenberg** [comment]
Done in the above commit. @terracoda can you please have another listen? I would expect all to sound as they did in the scene summary now.

If this sounds good to you we can make a new common code component for list items with this fix so that this is applied to all sims. If you think it doesn't sound as good we can revert the change.

**Comment by terracoda** [comment]
Sorry @jessegreenberg, it was the lists in the Play Area that still had their counting semantics, and the list in the Scene Summary that did not.

Now none of the lists have their counting semantics.

Before the last commit, the mentioning of the bullets had been removed from all lists, and  the counting semantics were still on the list items in the Play Area.

VO would say:
heading level 3, mass 1, Blue Sphere 
    Force arrow is tiny, and points directly at mass 2.
    Size of force on mass 1 by mass 2 is 33.4 newtons. 2 of 3
    mass 1 is 2 billion kilograms, half the size of mass 2, and also 4 kilometers from mass 2. 3 of 3

So, now the "2 of 3" and "3 of 3" counting is gone on all lists.

Can you revert?
The `css` being applied to the Play Area lists was the preferred situation.

If I remember correctly from https://github.com/phetsims/a11y-research/issues/158, I think there are 2 things that have to happen:
1. remove the bullets with `css`, so VO doesn't say "bullet, bullet" on every list item
2. add an explicit list role to the parent `ul` because Apple erroneously removes the role when the bullets are removed with `css`.

Is this what you have done @jessegreenberg ?
```html
<style>
  .hidden-list {
    list-style: none;
  }
</style>
<ul class="hidden-list" role="list">
    <li> Force arrow is tiny, and points directly at mass 2.</li>
    <li>Size of force on mass 1 by mass 2 is 33.4 newtons. </li>
    <li>mass 1 is 2 billion kilograms, half the size of mass 2, and also 4 kilometers from mass 2.</li>
</ul>
```
What code was applied to the Play Area lists before the last commit because they sounded better before the last commit?

**Comment by jessegreenberg** [comment]
Ah thanks for the reminder! No, I hadn't used `role=list` for any of the attempts in this issue so far.

> Before the last commit, the mentioning of the bullets had been removed from all lists,

Before the last commit there was no change made to the Play Area lists. Does that mean that Apple fixed this issue and this work is no longer necessary?? I would be curious to try with role=list applied and also with these changes reverted. If it sounds good without, I would prefer to remove this workaround.

**Comment by jessegreenberg** [comment]
In this version, we have `role="list"` and `list-style=none;` on the `ul`s.
https://phet-dev.colorado.edu/html/gravity-force-lab-basics/1.1.0-dev.40/phet/gravity-force-lab-basics_en_phet.html

In this version we have all this work removed.
https://phet-dev.colorado.edu/html/gravity-force-lab-basics/1.1.0-dev.39/phet/gravity-force-lab-basics_en_phet.html


@terracoda can you please have a listen to both of these and let me know which you prefer? I think there is a chance Apple has improved VoiceOver based on what you mentioned while testing the previous version and it is possible we can remove this CSS workaround.

**Comment by terracoda** [comment]
Oh, interesting, let’s revert then and have another listen.

**Comment by terracoda** [comment]
OK, I have listened to both versions 1.1.0-dev.39 (no fix applied) and 1.1.0-dev.40 (with special fix applied.

I tested in VO using 3 browsers Safari, Firefox and Chrome.

While each browser does something slightly differently (and Chrome sounds the best), they each do their thing consistently with either version. There is no difference in the versions, and thus no need for the special fix.

Apple has indeed (and thankfully) fixed the issue, VoiceOver no longer says "bullet, bullet" when reading out list items.

For the record, I also tested GFLB current published version and RIAW published version to see if I heard "bullet, bullet", and I do NOT! I am going to close this issue and a11y-research#158.

**Comment by terracoda** [comment]
Oh, sorry @jessegreenberg, you probabaly have to get rid of some unnecessary code. Leaving this for you to clean up and close.

**Comment by terracoda** [comment]
Oh, I just tested on MacOS High Sierra, and the above fix in version 1.1.0-dev.40 does indeed remove the "bullet", and keeps the list counting semantics.

So this fix does INDEED improve the experience on older systems.
I'm not sure how PhET normally would handle an improvement that improves things on older OS's.

In our case, since Apple has fixed this problem naturally in their latest OS, I would recommend not applying the fix. I think this would keep the code cleaner and more future compatible.

**Comment by jessegreenberg** [comment]
> Apple has indeed (and thankfully) fixed the issue, VoiceOver no longer says "bullet, bullet" when reading out list items.

Great news, thanks @terracoda!

> you probably have to get rid of some unnecessary code. 

Yep, I will go through and remove the changes made in this issue.

>  I just tested on MacOS High Sierra, and the above fix in version 1.1.0-dev.40 does indeed remove the "bullet", and keeps the list counting semantics.

Good to know, I guess that confirms that this would have been a good fix if we needed it. I agree that we shouldn't include this code, making sure that Interactive Description works on the latest AT is in line with what we have done historically.

**Comment by terracoda** [comment]
I closed phetsims/a11y-research#158, too and referenced this issue.
